src/DataTable/__tests__/DataTable.test.tsx

```diff
@@ -262,4 +272,259 @@ describe('DataTable', () => {
+          data={[
+            {
+              id: 1,
+              value: 1,
+            },
+            {
+              id: 2,
+              value: 2,
+            },
+            {
+              id: 3,
+              value: 3,
+            },
+          ]}
+          columns={[
+            {
+              header: 'Value',
+              field: 'value',
+              sortBy: true,
+            },
+          ]}
+          initialSortColumn="value"
+          initialSortDirection="ASC"
+        />,
+      )
+
+      const header = screen.getByRole('columnheader', {
+        name: 'Value',
+      })
+      expect(header).toHaveAttribute('aria-sort', 'ascending')
+
+      const rows = screen
+        .getAllByRole('row')
+        .filter(row => {
+          return queryByRole(row, 'cell')
+        })
+        .map(row => {
+          const cell = getByRole(row, 'cell')
+          return cell.textContent
+        })
+      expect(rows).toEqual(['1', '2', '3'])
+    })
+
+    it('should change the sort direction on mouse click', async () => {
+      const user = userEvent.setup()
+      render(
+        <DataTable
+          data={[
+            {
+              id: 1,
+              value: 1,
+            },
+            {
+              id: 2,
+              value: 2,
+            },
+            {
+              id: 3,
+              value: 3,
+            },
+          ]}
+          columns={[
+            {
+              header: 'Value',
+              field: 'value',
+              sortBy: true,
+            },
+          ]}
+          initialSortColumn="value"
+          initialSortDirection="ASC"
+        />,
+      )
+
+      function getRowOrder() {
+        return screen
+          .getAllByRole('row')
+          .filter(row => {
+            return queryByRole(row, 'cell')
+          })
+          .map(row => {
+            const cell = getByRole(row, 'cell')
+            return cell.textContent
+          })
+      }
+
+      expect(getRowOrder()).toEqual(['1', '2', '3'])
+
+      // Transition from ASC -> DESC order
+      await user.click(screen.getByText('Value'))
+      expect(getRowOrder()).toEqual(['3', '2', '1'])
+
+      // Transition from DESC -> ASC order
+      await user.click(screen.getByText('Value'))
+      expect(getRowOrder()).toEqual(['1', '2', '3'])
+    })
+
+    it('should change the sort direction on keyboard Enter or Space', async () => {
+      const user = userEvent.setup()
+      render(
+        <DataTable
+          data={[
+            {
+              id: 1,
+              value: 1,
+            },
+            {
+              id: 2,
+              value: 2,
+            },
+            {
+              id: 3,
+              value: 3,
+            },
+          ]}
+          columns={[
+            {
+              header: 'Value',
+              field: 'value',
+              sortBy: true,
+            },
+          ]}
+          initialSortColumn="value"
+          initialSortDirection="ASC"
```

Comment:
What happens if `initialSortColumn` is set to a column that doesn't have `sortBy: true`?

Reply:
@colebemis right now it will assume that it is sortable but it should 100% be validating this 👍 

I created this issue to track all the combinations and what the outcome should be: https://github.com/github/primer/issues/1854

Let me know if I missed any scenarios here! I'll tackle this in a follow-up PR